### PR TITLE
Benchmarks

### PR DESCRIPTION
**Pull request by gasse**
- [x] JSON serializable `Benchmark` class with `name`, `high_level_action_set` and `env_args_list`
    ```python
    from dataclasses import dataclass
    
    from dataclasses_json import DataClassJsonMixin
    
    
    @dataclass
    class Example(DataClassJsonMixin):
        a: int
        b: str
    
        def do_something(self):
            print(self.a, self.b)
      
    x = Example(0, "hello")
    x.do_something()
    
    x_json = x.to_json()
   
    y = Example.from_json(x_json)
    y.do_something()
    ```
 - [x] flexible, easy to crate your own benchmark
    ```python
    Benchmark(
        name="miniwob_tiny_test",
        high_level_action_set=DEFAULT_HIGHLEVEL_ACTION_SETS["miniwob"],
        env_args_list=_make_env_args_list_from_repeat_tasks(
            task_list=["miniwob.click-dialog", "miniwob.click-checkboxes"],
            max_steps=5,
            n_repeats=2,
            seeds_rng=np.random.RandomState(42),
        ),
    )
    ```
 - [x] CSV files with benchmark task list and their metadata
 - [x] default benchmarks collection
   - [x] miniwob_all
   - [x] miniwob_webgum
   - [x] miniwob_tiny_test
   - [x] miniwob_train
   - [x] miniwob_test
   - [x] webarena
   - [x] visualwebarena
   - [x] workarena_l1
   - [x] workarena_l1_sort
   - [x] workarena_l2
   - [x] workarena_l3

Missing:
 - [x] unit tests
 - [x] workarena l2 and l3 task metadata
 - [x] webarena metadata (categories)
 - [x] visualwebarena task list and metadata
 - [ ] integration in AgentLab (different PR)

### CODE REVIEW

**Comment by gasse** [comment]
solves #170 #160

File: browsergym/experiments/src/browsergym/experiments/benchmark.py [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+            strict=self.strict,
+            retry_with_force=self.retry_with_force,
+            demo_mode=self.demo_mode,
+        )
+
+
+@dataclass
+class Benchmark(DataClassJsonMixin):
+    name: str
+    high_level_action_set: HighLevelActionSetArgs
```
Comment (recursix) [inline browsergym/experiments/src/browsergym/experiments/benchmark.py:41]: should we add `_args`?

File: browsergym/experiments/src/browsergym/experiments/benchmark.py [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+        subsets=["chat", "infeas", "bid", "tab", "nav"],
+        multiaction=False,
+        strict=False,
+        retry_with_force=False,
+        demo_mode="off",
+    ),
+}
+
+BENCHMARKS = {
+    "miniwob_all": lambda: Benchmark(
```
Comment (recursix) [inline browsergym/experiments/src/browsergym/experiments/benchmark.py:117]: put a commont to reassure the surprised user that there is a lambda

File: browsergym/experiments/src/browsergym/experiments/benchmark.py (hunk 1 above)
Comment (gasse) [reply browsergym/experiments/src/browsergym/experiments/benchmark.py:41]: done

File: browsergym/experiments/src/browsergym/experiments/benchmark.py (hunk 2 above)
Comment (gasse) [reply browsergym/experiments/src/browsergym/experiments/benchmark.py:116]: done

**Comment by gasse** [comment]
Trust-approved by @recursix

**Pull request merged**